Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+
+#include "MultiplayerConnectionViewportMessageSystemComponent.h"
+#include <Atom/RPI.Public/ViewportContextBus.h>
+#include <Atom/RPI.Public/ViewportContext.h>
+#include <AzCore/Console/IConsole.h>
+#include <AzFramework/Entity/EntityDebugDisplayBus.h>
+#include <AzNetworking/Framework/INetworking.h>
+#include <Multiplayer/IMultiplayerSpawner.h>
+#include <Multiplayer/MultiplayerConstants.h>
+
+namespace Multiplayer
+{
+    constexpr float defaultConnectionMessageFontSize = 0.7f;
+    const AZ::Vector2 viewportConnectionBottomRightBorderPadding(-40.0f, -40.0f);
+
+    AZ_CVAR_SCOPED(bool, cl_viewportConnectionStatus, true, nullptr, AZ::ConsoleFunctorFlags::DontReplicate,
+        "This will enable displaying connection status in the client's viewport while running multiplayer.");
+
+    AZ_CVAR_SCOPED(float, cl_viewportConnectionMessageFontSize, defaultConnectionMessageFontSize, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, 
+        "The font size used for displaying updates on screen while the multiplayer editor is connecting to the server.");
+    
+    void MultiplayerConnectionViewportMessageSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<MultiplayerConnectionViewportMessageSystemComponent, AZ::Component>()
+                ->Version(1);
+        }
+    }
+    
+    MultiplayerConnectionViewportMessageSystemComponent::MultiplayerConnectionViewportMessageSystemComponent()
+    {
+        AZ::Interface<IMultiplayerConnectionViewportMessage>::Register(this);
```

Comment:
You can derive from ::Registrar to achieve the same thing.

Reply:
that's awesome, thank you! fixed!